app/models/action/edition_diff_factory.rb

```diff
@@ -0,0 +1,45 @@
+class Action::EditionDiffFactory
+  attr_reader :previous_draft_edition, :updated_draft_edition
+
+  def add_previous_draft_edition(previous_draft_edition)
+    @previous_draft_edition = get_edition_attributes(previous_draft_edition)
+  end
+
+  def add_updated_draft_edition(updated_draft_edition)
+    @updated_draft_edition = get_edition_attributes(updated_draft_edition)
+  end
+
+  def create_draft_edition_diff
+    # we don't raise an error if there is no @previous_draft_edition present,
+    # as it's possible for no previous draft edition to exist
+    raise Action::NoUpdatedDraftEditionProvidedError unless @updated_draft_edition.present?
+    create_edition_diff(previous_draft_edition, updated_draft_edition)
+  end
+
+private
+
+  def create_edition_diff(edition_a, edition_b)
+    HashDiff.diff(edition_a, edition_b)
+  end
+
+  def get_edition_attributes(edition)
+    attributes = edition.try(:attributes)
```

Comment:
be good to not include timestamps here. Rather than using attributes could loop through [TOP_LEVEL_FIELDS](https://github.com/alphagov/publishing-api/blob/master/app/models/edition.rb#L11) to call the methods for each - as there's some subtle differences between the attributes and method versions (SymbolizeJSON)